docs/DeveloperGuide.md

```diff
@@ -4,10 +4,137 @@
 
+Given below is a quick overview of main components and how they interact with each other.
+
+**`Main`** is in charge of the app launch, shut down, and taking user input.
+
+* At the app launch, it loads the data from the file system, and initializes other components.
+* At the app shut down, it saves the data to the file system, and shuts down other components.
+
+The app's work is done by the following components:
+
+* [**`MainLogic`**](#sublogic-component): The main logic command executor.
+* [**`SubLogic`**](#sublogic-component): The sub logics(i.e. **`MenuLogic`**, **`OrderLogic`**) command executor.
+* [**`Parser`**](#parser-component): The parser that parses user input to command.
+* [**`Model`**](#model-component): The data model that stores the data.
+* [**`Command`**](#command-component): Represents a command that the user can execute.
+* [**`Storage`**](#storage-component): Reads data from, and writes data.
+
+**How the architecture components interact with each other**
+
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues 
+the command `Create order -menu 01`, `add -item 001` and `complete`.
+
+![Sequence Diagram](images\ArchitectureSequanceDiagram.png)
+
+### UI Component
+The UI component is responsible for parsing user input into commands that can be executed by the logic component.
+There is a `analyzeInput` method in the `Parser` class that interprets the user input and classifies 
+it into a `CommandType` enum.<br>
+The `splitInput` method is used to split the user input into an array of strings, according to the command type.
+It returns an array containing any arguments that are needed to execute the command.
+
+### Logic Component
+The logic component consists of classes that handle the logic of the application. The logic component is divided into
+`MainLogic` and `SubLogic` which consists of `OrderLogic` and `MenuLogic`.
+
+* [**`MainLogic`**](#model-component): A class to handle the first level commands, and pass the user input to corresponding
+classes for analysis and execution.
+* [**`OrderLogic`**](#model-component) and [**`MenuLogic`**](#model-component): A class to handle the second level commands,
+and pass the user input to corresponding classes for analysis and execution.
+
+### Command Component
+The command component consists of two different command interfaces: `MainCommand` and `OrderCommand`. The `MainCommand`
+interface is for the various command classes that are used in the `MainLogic`, while the `OrderCommand` interface is for
+command classes used in `OrderLogic`.  
+
+A `Command` object will be created in either the `MainLogic` or `OrderLogic`
+based on what command the user has inputted(e.g., `MainHelpCommand` object is created when the user inputs the `help`
+command). The `execute()` method of the `Command` object is then called to execute the command, which may require
+certain arguments based on the type of command.
```

Comment:
See nit above (similar to UI component).
Alternatively, a diagram may be useful to make it more readable/ consistent